spec/linter-luacheck-spec.js

```diff
@@ -0,0 +1,76 @@
+'use babel';
+
+import { join } from 'path';
+import { beforeEach, it } from 'jasmine-fix'; // eslint-disable-line import/no-extraneous-dependencies
+import linterLuacheck from '../lib/init';
+
+const emptyPath = join(__dirname, 'fixtures', 'empty.lua');
+const goodPath = join(__dirname, 'fixtures', 'good.lua');
+const warningPath = join(__dirname, 'fixtures', 'warning.lua');
+const errorPath = join(__dirname, 'fixtures', 'error.lua');
+
+const errorStr = "expected identifier near 'local'";
+const warningStr = "unused argument 'self'";
+
+describe('The luacheck provider for Linter', () => {
+  const lint = linterLuacheck.provideLinter().lint;
+
+  beforeEach(async () => {
+    atom.workspace.destroyActivePaneItem();
+    await atom.packages.activatePackage('linter-luacheck');
+  });
+
+  it('should be in the packages list', () =>
+    expect(atom.packages.isPackageLoaded('linter-luacheck')).toBe(true)
+  );
+
+  it('should be an active package', () =>
+    expect(atom.packages.isPackageActive('linter-luacheck')).toBe(true)
+  );
+
+  it('finds nothing wrong with an empty file', async () => {
+    const editor = await atom.workspace.open(emptyPath);
+    const result = await lint(editor);
+
+    expect(result.length).toEqual(0);
+  });
+
+  it('finds nothing wrong in good.lua', async () => {
+    const editor = await atom.workspace.open(goodPath);
+    const result = await lint(editor);
+
+    expect(result.length).toEqual(0);
+  });
+
+  it('finds warning in warning.lua', async () => {
+    const editor = await atom.workspace.open(warningPath);
+    const result = await lint(editor);
+
+    expect(result.length).toEqual(1);
+    expect(result[0].severity).toBeDefined();
+    expect(result[0].severity).toEqual('warning');
```

Comment:
Sure, thought I had a comment in there already to that effect but it looks like that got dropped.